cmd/skopeo/utils_test.go

```diff
@@ -1,68 +1,92 @@
 package main
 
 import (
-	"flag"
 	"os"
 	"testing"
 
 	"github.com/containers/image/v5/types"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
 // fakeGlobalOptions creates globalOptions and sets it according to flags.
-// NOTE: This is QUITE FAKE; none of the urfave/cli normalization and the like happens.
-func fakeGlobalOptions(t *testing.T, flags []string) *globalOptions {
-	app, opts := createApp()
-
-	flagSet := flag.NewFlagSet(app.Name, flag.ContinueOnError)
-	for _, f := range app.Flags {
-		f.Apply(flagSet)
-	}
-	err := flagSet.Parse(flags)
+func fakeGlobalOptions(t *testing.T, flags []string) (pflag.FlagSet, *globalOptions) {
+	opts := globalOptions{}
+	fs := pflag.FlagSet{}
+	fs.BoolVar(&opts.debug, "debug", false, "enable debug output")
+	fs.BoolVar(&opts.tlsVerify, "tls-verify", true, "Require HTTPS and verify certificates when accessing the registry")
+	fs.StringVar(&opts.policyPath, "policy", "", "Path to a trust policy file")
+	fs.BoolVar(&opts.debug, "insecure-policy", false, "run the tool without any policy check")
+	fs.StringVar(&opts.registriesDirPath, "registries.d", "", "use registry configuration files in `DIR` (e.g. for container signature storage)")
+	fs.StringVar(&opts.overrideArch, "override-arch", "", "use `ARCH` instead of the architecture of the machine for choosing images")
+	fs.StringVar(&opts.overrideOS, "override-os", "", "use `OS` instead of the running OS for choosing images")
+	fs.DurationVar(&opts.commandTimeout, "command-timeout", 0, "timeout for the command execution")
+	fs.StringVar(&opts.registriesConfPath, "registries-conf", "", "path to the registries.conf file")
```

Comment:
Defining a completely new set of options here makes the tests dubious useful for testing CLI option handling of the actual code.